.github/workflows/tests.yml

```diff
@@ -37,11 +37,55 @@ jobs:
 
     - run: make install version
     - run: make unit-tests
-    - run: make integration-tests PYTEST_ARGS="--random-order"
+      env:
+        COVERAGE_FILE: ".coverage.${{ matrix.python-version }}"
+        PYTEST_ARGS: "--cov=karapace --cov-append"
+    - run: make integration-tests
+      env:
+        COVERAGE_FILE: ".coverage.${{ matrix.python-version }}"
+        PYTEST_ARGS: "--cov=karapace --cov-append --random-order"
 
     - name: Archive logs
       uses: actions/upload-artifact@v4
       if: ${{ always() }}
       with:
         name: karapace-integration-test-logs-${{ matrix.python-version }}
         path: /tmp/ci-logs
+    - name: Archive coverage file
+      uses: actions/upload-artifact@v4
+      with:
+        name: "coverage-${{ matrix.python-version }}"
+        path: ".coverage.${{ matrix.python-version }}"
+
+  coverage:
+    name: Coverage report
+    runs-on: ubuntu-latest
+    needs: tests
+    permissions:
+      pull-requests: write
+      contents: write
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Download coverage
+      id: download_coverage
+      uses: actions/download-artifact@v4
+      with:
+        pattern: coverage-*
+        merge-multiple: true
+
+    - run: make karapace/version.py
+
+    - name: Post coverage comment
+      id: post_coverage_comment
+      uses: py-cov-action/python-coverage-comment-action@v3
+      with:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        MERGE_COVERAGE_FILES: true
+
+    - name: Store PR coment to be posted
```

Comment:
Since you'd remove a commit, there's a tiny typo here: `coment`

Reply:
Done.